\chapter{Seasonal succession of free-living bacterial communities in coastal waters of the Western Antarctic Peninsula}\label{ch:swi}

\chapterdisclaimer{This chapter is adapted from the following publication:\\
Luria, C. L., Amaral-Zettler, L. A., Ducklow, H. W., and Rich, J. J. (2016). Seasonal succession of free-living bacterial communities in coastal waters of the Western Antarctic Peninsula. \emph{Frontiers in Microbiology} 7, 1731.\\
\\
Catherine Luria, Linda Amaral-Zettler, Hugh Ducklow, and Jeremy Rich designed and conducted the study. Linda Amaral-Zettler and Jeremy Rich contributed sequence data for the study. Catherine Luria analyzed data and wrote the manuscript with guidance from Jeremy Rich. Linda Amaral-Zettler, Hugh Ducklow, and Jeremy Rich provided input and edited the manuscript.
}


\section{Abstract}\label{sc:abstract2}
The marine ecosystem along the Western Antarctic Peninsula (WAP) undergoes a dramatic seasonal transition every spring, from almost total darkness to almost continuous sunlight, resulting in a cascade of environmental changes, including phytoplankton blooms that support a highly productive food web. Despite having important implications for the microbial loop and the biological carbon pump, little is known about how these changes impact bacterial succession in this region. Using 16S rRNA gene amplicon sequencing, we measured changes in free-living bacterial community composition and richness during a nine-month period that spanned winter to the end of summer. Chlorophyll \emph{a} concentrations were relatively low until summer when a major phytoplankton bloom occurred, followed three weeks later by a high peak in bacterial production. Richness in bacterial communities varied between \textasciitilde{}1,200-1,800 observed operational taxonomic units (OTUs) before the major phytoplankton bloom (out of \textasciitilde{}43,000 sequences per sample). During peak bacterial production, OTU richness decreased to \textasciitilde{}700 OTUs. The significant decrease in OTU richness only lasted a few weeks, after which time OTU richness increased again as bacterial production declined towards pre-bloom levels. OTU richness was negatively correlated with bacterial production and chlorophyll \textit{a} concentrations. Unlike the temporal pattern in OTU richness, community composition changed from winter to spring, prior to onset of the summer phytoplankton bloom. Community composition continued to change during the phytoplankton bloom, with increased relative abundance of several taxa associated with phytoplankton blooms, particularly \textit{Polaribacter}. Bacterial community composition began to revert toward pre-bloom conditions as bacterial production declined. Overall, our findings clearly demonstrate the temporal relationship between phytoplankton blooms and seasonal succession in bacterial growth and community composition. Our study also highlights the importance of high-resolution time series sampling, especially during the relatively under-sampled winter and spring, which enabled us to discover seasonal changes in bacterial community composition that preceded the summertime phytoplankton bloom.

\section{Introduction}\label{introduction2}

Strong seasonal patterns in the marine ecosystem west of the Antarctic Peninsula (WAP) provide a natural experiment to assess how bacteria respond over time to changes in both biotic and abiotic factors. The Antarctic spring brings about a cascade of environmental changes, including light-driven modification of dissolved organic matter (DOM), sea ice melting and retreat, warmer water temperatures and stratification of the water column. These physical changes trigger phytoplankton blooms that support large stocks of upper level consumers \citep{Smetacek2005-tz} and represent a potentially important sink for atmospheric \ce{CO2} \citep{Arrigo2008-xp}.

In the water column, some bacterial taxa are adapted to colonize and attach to particles through surface adhesion and gliding motility, while other taxa are more adapted towards life as free-living cells \citep{dang2016}. This is reflected in differences in community composition between the particle attached and free-living communities \citep{delong1993phylogenetic, Ortega-Retuerta2013freeliving, rieck2015freeliving}. Particle attached bacteria play an important role in the initial degradation of particulate matter, hydrolyzing polymers and releasing smaller molecules that can diffuse away from particles and be utilized by free-living bacteria \citep{dang2016}. While particle attached bacteria may have higher specific activity, free-living cells generally contribute more to overall bacterial activity in the water column due to greater overall cell abundance, but there are exceptions \citep{iriberri1987, turley1994freeliving, rieck2015freeliving}. Therefore, studies that focus on free-living cells describe an important part of the bacterial community but not the entire community \citep{Teeling2012-jz, williams2013role, Sunagawa1261359}.

Bacterial interactions with phytoplankton contribute to ecosystem function in multiple ways in both the WAP and the global ocean \citep{Cole1982-yp, Croft2005-em, Sher2011-op}. Sustained primary production is partly dependent upon the microbial loop, in which particle attached and free-living bacteria degrade DOM and are consumed by bacterivores, thereby recycling nutrients \citep{Azam1983-wo}. Studies in marine ecosystems indicate that bacterial growth, in turn, is frequently dependent on phytoplankton-derived dissolved organic matter (DOM) \citep{Church2000-wc, Moran2001-ot, Morn2002-cx, Piquet2011-ot, dsvse12, Kim2014-oj}. Phytoplankton-derived carbon likely influences the succession of bacterial communities as various bacterial taxa differ in their ability to degrade phytoplankton-derived DOM and particulate detritus \citep{Kerkhof1999-nr, Pinhassi2004-kc, Teeling2012-jz}. Lower bacterial diversity, in terms of both richness and evenness, accompanies seasonal changes, reflecting an increase in abundance of relatively few bacterial taxa \citep{Gilbert2012-ta, Ladau2013-ro}.

Certain groups of bacteria (e.g., Flavobacteria and Rhodobacteraceae) increase in abundance during phytoplankton blooms, while other groups such as \emph{Pelagibacter} are better adapted to the free-living state non-bloom conditions \citep{williams2013role, Buchan2014-yh, Voget2015-ch}. Flavobacteria, described as `first responders' to phytoplankton blooms, break down complex organic matter by direct attachment and exoenzymatic attack of phytoplankton cells and phytoplankton-derived detrital particles \citep{williams2013role}. An abundant genus of Flavobacteria in the bacterioplankton is \emph{Polaribacter}, which possess traits consistent with a life-strategy of particle attachment and polymer degradation \citep{Fernandez-Gomez2013-wr}. However, \emph{Polaribacter} is metabolically flexible and also abundant in the free-living community, indicating that its ecological niche likely extends beyond particle attachment \citep{williams2013role, smith2013freeliving}. Flammeovirgaceae, a family within Bacteroidetes, has also been associated with the degradation of algal-derived polysaccharides \citep{Chan2015-pw, Liu2015-po, Voget2015-ch}. Members of Rhodobacteraceae are often found in close association with phytoplankton blooms in either the particle attached or free-living fraction of the community. They generally utilize small molecular weight substrates, including the degradation products produced by Flavobacteria \citep{Pinhassi2004-kc, West2008-vp, wemheuer2015green}.

Studies of bacterial seasonal succession have emphasized the role of bacteria as degraders of labile organic matter \citep{Teeling2012-jz, Moran2015-dd, Needham2016-xb}. However, there is increasing evidence that bacterial interactions with phytoplankton may influence the development of phytoplankton blooms themselves through bacterial production of key vitamins, chelating agents, or hormones that stimulate or impede phytoplankton growth \citep{Amin2012-tv, Amin2015-pp, Prieto2015-oi, Wang2016-lt}. Bacterial succession prior to the onset of phytoplankton blooms has been arguably under-studied \citep{Moran2015-dd, Needham2016-xb}.

Previous analyses in the WAP, based either on community fingerprinting techniques (i.e. denaturing gradient gel electrophoresis) over one or more seasons, or on high-throughput DNA sequencing from limited mid-winter and mid-summer sampling dates, hint at a relationship between bacterial community succession and phytoplankton blooms similar to that observed in more temperate regions \citep{mpmtbwd98,mg07,grwddecm12, Luria2014-dj}. However, the intervening time period between winter and summer is severely under-sampled in the WAP, as it is throughout the Southern Ocean, making comparisons to other systems difficult. Our objective was to obtain a new high-resolution seasonal time-series of bacterial properties to determine how free-living bacterial succession proceeds during the dynamic Antarctic winter-to-summer transition. We hypothesized that a phytoplankton bloom would trigger bacterial succession as has been demonstrated previously in temperate regions (e.g., \citealt{Teeling2012-jz}). Our findings on bacterial community succession provide new evidence of coupling between phytoplankton and bacterial blooms.



\section{Materials and Methods}\label{materials-and-methods2} 

\subsection{Field sampling and contextual data}\label{field-sampling-and-contextual-data} 
Seawater samples were collected from coastal surface waters at Palmer Station, on the west coast of the Antarctic Peninsula at one to two week intervals beginning in the austral mid-winter (July 2013) and ending in late summer (March 2014). Samples were drawn directly from a seawater intake located at a depth of \SI{6}{\m}, \SI{16}{\m} from the station. Triplicate 20-L samples were collected in acid-washed Nalgene carboys and immediately transferred to the laboratory for processing. All processing took place in a 0\textdegree C cold room to maintain initial water temperature.

Each 20-L carboy was sub-sampled for dissolved nutrient (phosphate, silicate, nitrite and nitrate), particulate organic carbon and nitrogen (POC and PON), chlorophyll \textit{a} (chl \textit{a}), and bacterial abundance and bacterial production measurements. Samples were collected and processed according to Palmer LTER standard protocols (\url{http://oceaninformatics.ucsd.edu/datazoo/data/pallter/datasets}). Briefly, nutrient samples were filtered through combusted \SI{0.7}{\micro\meter} glass fiber filters (Whatman, GE Healthcare Life Sciences, Piscataway, NJ) and frozen at -80\textdegree C until analysis on a SEAL AutoAnalyzer 3 \citep{ducklow_doi_2016a}. POC and PON samples were collected on combusted \SI{0.7}{\micro\meter} glass fiber filters from 1-3 L seawater and were frozen at -80\textdegree C until analysis via combustion using a Perkin Elmer 2400 Series II CHNS/O Analyzer. Chl \textit{a} samples, as with POC and PON, were collected on \SI{0.7}{\micro\meter} glass fiber filters from 1-3 L seawater and were assayed fluorometrically using acetone extracts \citep{schofield_doi_2016}. Bacterial abundance samples were analyzed by flow cytometry following the protocol of \citet{Gasol2000-mn}, with SYBR\textsuperscript{\textregistered} Green I nucleic acid staining (Invitrogen, Carlsbad, CA) on an Accuri C6 flow cytometer (BD Biosciences, San Jose, CA) \citep{ducklow_doi_2016b}. Bacterial production rates were derived from rates of \ce{^3H}-leucine incorporation, as previously described \citep{dsvse12,ducklow_doi_2016b}. 

Samples for bacterial community composition were collected via gentle vacuum filtration (-0.5 bar vacuum) of approximately 2-6 L of seawater through successive \SI{3.0}{\micro\meter} polycarbonate (EMD Millipore, Billerica, MA) and \SI{0.22}{\micro\meter} polyethersulfone (EMD Millipore, Billerica, MA) filters. Filters were flash-frozen with liquid \ce{N2} and stored at -80\textdegree C until further processing. 

\subsection{16S rRNA gene library generation }\label{s-rrna-gene-library-generation} 

Bacteria that passed through \SI{3.0}{\micro\meter} pore size filters and were retained on \SI{0.22}{\micro\meter} pore size filters were analyzed, and therefore our focus is on the free-living bacterial community composition and not the particle attached component. DNA was extracted from cells captured on the \SI{0.22}{\micro\meter} pore size filters using a DNeasy Plant Mini Kit (Qiagen, Valencia CA) with an additional bead-beating step. Initially, each filter was cut into small pieces and placed into a tube along with \SI{0.1}{\micro\meter} diameter silica beads. After incubation with AP1 and RNase as in the manufacturer's protocol, samples were vigorously vortexed for 1 minute to lyse cells. DNA extraction was then completed according to the manufacturer's protocol. 

For each DNA sample, the V6 hypervariable region of the 16S rRNA gene was amplified in two stages following the protocol described previously in \citet{Eren2013-ob}. In the first stage, \textasciitilde{}100 bp of the V6 region were amplified in triplicate \SI{33}{\micro\liter} PCR reactions containing 1.0 unit of Platinum Taq Hi-Fidelity Polymerase (Life Technologies, Carlsbad CA), 1X Hi-Fidelity buffer, \SI{200}{\micro\Molar} dNTP PurePeak DNA polymerase mix (Pierce Nucleic Acid Technologies, Milwaukee, WI), 2.0 mM \ce{MgCl2}, 0.06\% BSA, \SI{0.2}{\micro\Molar} forward and reverse non-fusion primers \citep{Eren2013-ob}, and approximately 10 ng template DNA. The PCR reaction consisted of a 3-minute initial denaturation step at 94\textdegree C, 25 cycles of 94\textdegree C for 30s, 60\textdegree C for 45s, and 72\textdegree C for 60s, and a final 2-minute extension at 72\textdegree C. After amplification, triplicate samples were pooled and purified products using a MinElute Reaction Cleanup Kit (Qiagen, Valencia CA) with DNA finally eluted in \SI{10}{\micro\liter} of Qiagen EB buffer.

After quality checks using a Fragment Analyzer (Advanced Analytics, Ames IA), a second fusion PCR reaction was conducted with a set of custom fusion primers consisting of Illumina adaptors, 12 different inline barcodes (forward primers), 8 dedicated indices (reverse primers), and the same V6 primer sequences used in the first round of PCR. The PCR reaction conditions were similar to those described above except that only a single reaction was run for each sample with approximately 2-3 ng purified PCR product as template and 10 reaction cycles. After the final concentration of PCR products was determined using a Qubit 3.0 fluorometer with PicoGreen (LifeTechnologies, Carlsbad CA), equimolar amounts of each sample were pooled. The 200-240 bp fraction of the sample pool was selected on 1\% agarose using Pippin Prep (SageScience, Beverly MA) and the final DNA concentration was measured using qPCR (Kapa Biosystems, Woburn MA) prior to sequencing on one lane of an Illumina HiSeq 1000 cycle paired-end run.

\subsection{Data analysis}\label{data-analysis}

Low-quality sequences were filtered from the resulting data by discarding reads without 100\% consensus between forward and reverse paired-end sequencing reads \citep{Eren2013-ob}. Observed taxonomic units (OTUs) were clustered using Qiime's (v 1.9.1) open reference OTU picking with the default UCLUST method, a minimum cluster size of 2, and a 97\% similarity threshold and were assigned Greengenes taxonomy (version 13\_8; \citealt{Caporaso2010-ee, Edgar2010-gq, McDonald2012-qf}). After removing OTUs classified as chloroplasts, rarefied libraries were produced by randomly down-sampling to the smallest library size, 43,308 sequences. Subsequent analyses were based on rarefied libraries unless otherwise indicated.

Alpha diversity metrics included observed richness (number of OTUs observed in each library), Shannon's diversity, and Pielou's evenness, as well as non-parametric Chao-Jost richness estimates based on un-rarefied libraries as implemented in the R package \code{iNEXT} \citep{Chao2012-nw, chao2014rarefaction}. The beta diversity between samples was visualized with non-metric multidimensional scaling (NMDS) based on Bray-Curtis similarity using the \code{metaMDS} function in the \code{vegan} package for R \citep{Oksanen_2015}. NMDS was also used to assess several normalization methods for un-rarefied libraries: converting read numbers to relative abundances, as well as Relative Log Expression (RLE) and Trimmed Mean of M-Values (TMM) normalizations as implemented in \code{edgeR} \citep{McMurdie2013-bb}.

\begin{figure}[ht] 
\centering 
\includegraphics[width=0.6\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_1_k_selection}
\caption[Scree plot of number of k-means clusters versus within groups sum of squares.]{k-means analysis of Loess filtered OTUs. $k = 5$ clusters was manually chosen to partition non-overlapping clusters in the network analysis.} 
\label{fig:ch2:supp2_2} 
\end{figure}

To identify OTUs with non-random temporal patterns, we first selected OTUs with a relative abundance ${\geq}$ 0.01 in at least three samples. We assessed the seasonality of this subset through local polynomial regression (LOESS) with serial day as the independent variable and relative abundance as the dependent variable, using $r^{2} > 0.8$ as a threshold to identify OTUs with non-random temporal patterns. We used these OTUs to generate a co-occurrence network based on Pearson's correlation ($r > 0.5$) using the Cytoscape (version 3.4.0; \citealt{smobwr}) CoNet plugin (version 1.1.1.beta; \citealt{faust2012microbial}). We extracted non-overlapping clusters from this network through k-means clustering using the Cytoscape clusterMaker2 plugin (version 0.9.5; \citealt{morris2011clustermaker}) after first selecting a reasonable value for k through the evaluation of a scree plot of within-clusters sum of squares (Figure \ref{fig:ch2:supp2_2}). The resulting network clusters were visualized in Cytoscape. This helped us to identify groups of OTUs with similar temporal trends.


\afterpage{
\begin{landscape}
\begin{table}[htbp]
\centering
\caption[Univariate and stepwise linear regressions of bacterial production in relation to environmental factors.]{Univariate and stepwise linear regressions of bacterial production in relation to environmental factors. Each reported $r^2$ represents a separate regression analysis.}
\label{ch2:tab1}
\begin{tabular}{@{}lllll@{}}
\toprule
Environmental factor (EF)         & Measured univariate\textsuperscript{a} $r^2$         & Interpolated univariate\textsuperscript{b} $r^2$        & Stepwise lags\textsuperscript{c} (days)       & Stepwise $r^2$       \\ \midrule
Temperature                       & 0.46                                                & 0.59                                                   & 0, 10                                        & 0.68                 \\
Phosphate                         & 0.49                                                & 0.75                                                   & NSd                                          & NS                   \\
Silicate                          & 0.29                                                & 0.53                                                   & NS                                           & NS                   \\
Nitrate+nitrite                   & 0.52                                                & 0.77                                                   & NS                                           & NS                   \\
POC                               & 0.34                                                & 0.62                                                   & 0, 10, 20                                    & 0.79                 \\
PON                               & 0.31                                                & 0.62                                                   & 0, 20                                        & 0.69                 \\
Chl \emph{a}                             & 0.23                                                & 0.25                                                   & 0, 10, 20                                    & 0.93                 \\ \midrule
                                  &                                                     &                                                        &                                              &                      \\
\multicolumn{5}{l}{\textsuperscript{a} Univariate linear regression of measured variables ($p<10^{-5}$, degrees of freedom (df)=48,}                                                 \\
\multicolumn{5}{l}{except for chl \emph{a} where $p<10^{-7}$ and df=99).}                                                                                                         \\
\multicolumn{5}{l}{\textsuperscript{b} Univariate linear regression of interpolated variables ($p<10^{-15}$, df=162, except for chl \emph{a} where df=234).}                       \\
\multicolumn{5}{l}{\textsuperscript{c} Significant lags ($p<0.001$) in the final stepwise linear regression equations based on minimized AIC are shown}                              \\
\multicolumn{4}{l}{($p<10^{-15}$, df=162, except for chl \emph{a} where df=234).}                                                                                &                \\
\multicolumn{5}{l}{\textsuperscript{d} NS is not shown due to significant multicollinearity between different lags (VIF \textgreater 5).}                                                                                
\end{tabular}
\end{table}
\end{landscape}
}

We used univariate linear regression to assess the relationships between bacterial production and abundance and environmental predictors (temperature, inorganic nutrient levels, DOC, POC, and PON) using the \code{lm} function in R \citep{t08}. Additionally, we built distributed lag models in order to account for potentially delayed bacterial responses to changes in environmental parameters \citep{jorgenson1966rational}. In order to introduce lags to our data, we first used linear interpolation as implemented in the R package \code{zoo} to predict values between sampling dates and create regular time series \citep{zeileis2005zoo}. Interpolation added data between measured time points; therefore, results were generally more significant in the analysis of interpolated data compared to measured data, as indicated by $p$-values and some cases $r^2$-values (Table \ref{ch2:tab1}, compare measured univariate $r^2$ to interpolated univariate $r^2$). However, results for regression analyses were significant based on a cut-off of $p<10^{-5}$, whether measured or interpolated data were analyzed (Table \ref{ch2:tab1}). After testing different combinations of lag, we selected the following form for distributed lag models, where \emph{BR} is a bacterial response (i.e. production or abundance) and \emph{EF} is an environmental factor:

\emph{BR $\sim$ (EF with 0-day lag) + (EF with 10-day lag) + (EF with 20-day lag)}

From this initial equation, independent variables (i.e. \emph{EF} with different lags) were selected through stepwise linear regression based on minimizing Akaike information criterion (AIC) as implemented in the R packages \code{dynlm} and \code{MASS} \citep{venables2002modern,zeileis2016dynlm}. Variables were added and removed in order of greatest reduction in AIC in a stepwise manner (i.e. backward elimination and forward selection), thereby seeking an optimal balance between goodness of fit and parameterization. Models with multicollinearity, based on variance inflation factors (VIF) > 5, were discarded. Final linear regression models were assessed for: 1) fit, 2) significance, 3) VIF, 4) AIC, and 5) parsimony.

All of our sequence data are MIMARKS-compliant \citep{Yilmaz:2011aa} and will be deposited in the NCBI SRA. Copies of the MIMARKS table and chapter figures as well as the R code used for all data analysis and figure production can be accessed at: \url{https://github.com/cmluria/dissertation/Chapter_3_SWI}. 

\section{Results}\label{results}

Our sampling period encompassed changes in day length from \textasciitilde{}4 hours of sunlight when we began sampling in July to \textasciitilde{}22 hours of sunlight in December. Water temperatures reached a minimum of \hbox{-}1.1\textdegree C in August and a maximum of 1.6\textdegree C in January (Figure \ref{fig:ch2:supp2_4}). Sea ice cover in the Palmer region is variable from year to year; 2013 was a heavy sea ice year with dense sea ice cover during the winter that persisted at significant levels until December \citep{Stammerjohn2008-nj, massom2014state}. Chl \textit{a}, a proxy for phytoplankton biomass was low (\textless{}\SI{0.6}{\micro\gram\per\liter}) until October when a brief increase in chl \textit{a} occurred. The primary phytoplankton bloom, based on chl \textit{a}, peaked in early January, after which chl \textit{a} declined towards pre-bloom levels (Figure \ref{fig:chla_bp_richness}A). Dissolved inorganic nutrients (phosphate, silicate, and nitrate) were drawn down during the summer bloom, while particulate carbon and nitrogen increased, reflecting phytoplankton production (Figure \ref{fig:ch2:supp2_4}). Silicate drawdown was less than nitrate during the summer bloom, suggesting that other phytoplankton groups in addition to diatoms contributed to the bloom. Bacterial production was very low but still detectable from July to October (Figure \ref{fig:chla_bp_richness}B). By January, bacterial production increased substantially, peaking about 3 weeks after the peak in chl \textit{a} (Figure \ref{fig:chla_bp_richness}A, B). Our bacterial production rates measured from the seawater intake were similar to those measured at a near shore station (Figure \ref{fig:ch2:bpcomp}; data available at doi:10.6073/pasta/814628d18d4e23753d1164f0bd81095e). 

While several individual factors (phosphate, silicate, nitrate and nitrite, POC, and PON) were significantly correlated with bacterial production, stepwise linear regression revealed that chl \emph{a} with time lags of 0, 10, and 20 days was a very good predictor of both bacterial production ($r^2 = 0.93$) and abundance ($r^2 = 0.62$; Figure \ref{fig:supp2_6} and Table \ref{ch2:tab1}). Adding additional lag terms to this model did not improve the fit based on $r^2$. However, removing lag terms decreased the fit (e.g., Table \ref{ch2:tab1}, compare interpolated univariate $r^2$ to stepwise $r^2$). Although we tested regressions incorporating multiple environmental factors as independent variables, the resulting models did not improve upon the distributed lag model based on chl \emph{a} alone and furthermore, often displayed multicollinearity. 

\begin{figure}[ht!] 
\centering 
\includegraphics[width=\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_2_ancillary_data_noDOC} 
\caption[Contextual data for bacterial community composition time series.]{Contextual data associated with sequence data including (A) temperature, (B) nitrate and nitrite, (C) dissolved inorganic phosphate, (D) dissolved silicate, (E) bacterial cell abundance (BA), (F) particulate organic carbon, and (G) particulate organic nitrogen. Mean and standard error for each time point are shown.} 
\label{fig:ch2:supp2_4} 
\end{figure}

\begin{figure}[ht!] 
\centering 
\includegraphics[width=1.0\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_3_BP_Comparison_Across_Sites} 
\caption[Comparison of bacterial production in the Palmer Station seawater intake and at Palmer LTER Station B.]{A comparison of bacterial production from Palmer Station seawater intake and LTER Station B at 5-m and 10-m depths, December 2013-March 2014.} 
\label{fig:ch2:bpcomp} 
\end{figure}

\begin{figure}[ht!] 
\centering 
\includegraphics[width=0.6\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_4_BP_and_BA_models} 
\caption[Measured bacterial production and abundance and values predicted through a distributed lag model.]{Measured and modeled (A) bacterial production and (B) bacterial abundance. The model fit for both production and abundance is the stepwise linear regression equation of chlorophyll \emph{a} with 0-, 10-, and 20-day lags, with minimized AIC.} 
\label{fig:supp2_6} 
\end{figure}

\begin{figure}[ht!] 
\centering 
\includegraphics[width=0.6\textwidth]{Chapter_3_SWI/Figures/Figure_1_Chla_BP_richness} 
\caption[Changes in chlorophyll \emph{a}, bacterial production, and OTU richness over the nine-month time series.]{Characteristics of the summer phytoplankton bloom period (highlighted in grey), based on (A) increased chl \emph{a}, (B) increased bacterial production, and (C) decreased free-living bacterial OTU richness. Insets are shown corresponding to the time period before the summer phytoplankton bloom (A and B). Error bars represent standard error for each date.} 
\label{fig:chla_bp_richness} 
\end{figure}

In total, 68 samples, spread across 24 sampling dates, were sequenced, yielding 15 million short-read V6 16S rRNA gene sequences (\textasciitilde{}50,000-550,000 per library) corresponding to the free-living bacterial community, assigned to 28,857 OTUs. Given recent discussion regarding the statistical validity of library resampling \citep{McMurdie2014-iu}, we assessed different methods to characterize bacterial richness and community composition using un-rarefied sequence libraries. Although the method described by \citet{chao2014rarefaction} is intended to estimate true richness based on coverage regardless of sequencing depth, both OTU number ($p < 0.001, r^2 = 0.4)$ and estimated richness ($p <0.001, r^{2} = 0.35$) significantly correlated with library size (Figure \ref{fig:ch2:unrarrich}). This may indicate that the species abundance distributions vary widely among our datasets \citep{gwinn2015evaluating}. Likewise, NMDS based on un-rarefied sequence libraries showed the influence of initial library size even after relative abundance, RLE, and TMM normalizations (Figure \ref{fig:ch2:sixpanord}). For these reasons, all subsequent analyses relied on libraries rarefied to the smallest library size of 43,308 sequences.


\begin{figure}[ht!] 
\centering 
\includegraphics[width=0.7\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_5_unrarefied_estimated_richness} 
\caption[Observed and estimated (Chao-Jost) richness versus un-rarefied library size.]{Both (A) observed richness and (B) Chao-Jost estimated richness had significant, positive correlations with un-rarefied library size.} 
\label{fig:ch2:unrarrich} 
\end{figure}

\begin{figure}[ht!] 
\centering 
\includegraphics[width=\textwidth]{Chapter_3_SWI/Figures/Supplemental_Figure_6_Six_panel_ordination} 
\caption[Non-metric multidimensional scaling based on Bray-Curtis similarity matrices for un-rarefied libraries after three different normalization methods.]{Non-metric multidimensional scaling after running three different
normalization methods to accommodate size differences between un-rarefied libraries: simple conversion to relative abundance color-coded by month (A) and un-rarefied library size (B), Relative Log Expression (RLE) normalization (C-D), and Trimmed Mean of M-Values (TMM) normalization (E-F). Figures in left column contain samples color-coded by month. Figures in right column contain samples color-coded according to library size (number of sequences) prior to normalization.} 
\label{fig:ch2:sixpanord} 
\end{figure}


Observed richness for the free-living bacterial community varied significantly between midwinter and midsummer ($p < 0.0001$), ranging from a maximum of approximately 1800 OTUs in July to a minimum of approximately 700 OTUs in January (Figure \ref{fig:chla_bp_richness}C). On average, Chao-Jost estimated richness was four times greater than observed richness. Richness was negatively correlated with chl \emph{a} ($p<0.001, r^{2} = 0.16$) and bacterial production ($p<0.0001, r^{2} = 0.55$). A Tukey's HSD comparison across months confirmed that the decline in richness was significant in January and February ($p < 0.05$). Shannon Diversity and Pielou's evenness were also significantly lower in January ($p <0.0001$).


\begin{figure}[ht!] 
\centering 
\includegraphics[width=\textwidth]{Chapter_3_SWI/Figures/Figure_2_taxa_barplot} 
\caption[Taxonomic changes in bacterial community composition over the nine-month time series.]{Taxonomic changes in free-living bacterial community composition over the 9-month sampling period. Mean relative abundance in each month is shown. Each taxon contains one or more OTUs grouped according to taxonomic assignment to genus level or higher. The lowest taxonomic level available is shown.} 
\label{fig:taxa_barplot} 
\end{figure}

Figure \ref{fig:taxa_barplot} provides a broad overview of the changes in relative abundance of free-living bacterial taxa that occurred over the course of the season. Oceanospirillales and Pelagibacteraceae were the most abundant groups in winter and spring (July-November). SAR406, Rhodospirillales, and Deltaproteobacteria, including \textit{Nitrospina} and SAR324, were present at greater relative abundances in winter and early spring (July-September) than later in the field season. Changes in the relative abundances of taxonomic groups became most evident in January when Rhodobacteraceae had greater relative abundances than Pelagibacteraceae. \textit{Polaribacter} also increased significantly in January. We did not account for variation in 16S rRNA copy number and thus may have underestimated the abundances of lower copy number taxa like Pelagibacteraceae and overestimated the abundances of higher copy number taxa like Colwelliaceae. In addition to changes in relative abundances for broad taxonomic groups, co-occurrence network analysis with k-mean clustering revealed five temporal pattern types, with individual OTUs that peaked (i) during the winter or early spring, (ii) just prior to the summer bloom, (iii) very early in the summer bloom, (iv) at summer mid-bloom, or (v) after the summer bloom subsided (Figures \ref{fig:otu_succession} and \ref{fig:ch2_network_kmeans}). Some OTUs that were assigned the same family-level classification (e.g. Rhodobacteraceae) had distinct temporal patterns corresponding to different times during the summer bloom. 


\begin{figure}[htbp] 
\centering 
\includegraphics[width=\textwidth]{Chapter_3_SWI/Figures/Figure_3_OTU_succession_post_network_analysis} 
\caption[Five classes of successional patterns observed for individual OTUs over the nine-month time series.]{Successional patterns of free-living bacterial OTUs that peaked (A) during the winter or early spring, (B) prior to the summer bloom onset, (C) early in the summer bloom, (D) middle of the summer bloom, or (E) end of the summer bloom. Mean relative abundance and standard error for each date are shown. The affiliation of each OTU at the lowest taxonomic resolution available is shown.}
\label{fig:otu_succession} 
\end{figure}

Non-metric multidimensional scaling, based on Bray-Curtis similarities, demonstrated a shift in community composition from July through December before the onset of the summer phytoplankton bloom (Figure \ref{fig:ch2_nmds}). Community composition continued to change from December to January as the summer phytoplankton bloom developed. March samples clustered midway between the July-November and late January-early February samples, perhaps reflecting the beginning of a return to a pre-bloom community structure. PERMANOVA confirmed that community composition varied significantly between months ($p=0.$001). Bacterial communities fell into five groups based on hierarchical clustering of Bray-Curtis similarities. These groups corresponded to different time periods and ranges of bacterial production: July to November (winter and spring; \textless{} \SI{5}{\pico\mole \per\liter \per\hour}), December and March (before and after the summer bloom; \SIrange{5}{21}{\pico\mole \per\liter \per\hour}), early January (beginning of the summer bloom; \SIrange{26}{44}{\pico\mole \per\liter \per\hour}), late January to early February (summer mid-bloom; \SIrange{50}{129}{\pico\mole \per\liter \per\hour}), and late February (late in the summer bloom; \SIrange{17}{43}{\pico\mole \per\liter \per\hour}).

\begin{figure}[htbp] 
\centering 
\includegraphics[width=0.8\textwidth]{Chapter_3_SWI/Figures/Figure_4_otu_network_LoessPrefilter_kmeans_Pearsons} 
\caption[Co-occurrence networks of OTUs that displayed strong temporal patterns in relative abundance.]{Correlation network of free-living bacterial OTUs with strong temporal patterns in relative abundance, based on Pearson's correlation ($r > 0.5$) of LOESS-filtered OTUs. OTUs were partitioned into non-overlapping clusters using k-means clustering ($k = 5$).${~}$Edges represent positive correlations and nodes represent individual OTUs, labeled with OTU identification number and lowest level of taxonomic resolution available. The letters designate clusters that represented OTUs that peaked (A) during the winter or early spring, (B) prior to the summer bloom onset, (C) early in the summer bloom, (D) middle of the summer bloom, or (E) end of the summer bloom. Representative OTUs from each cluster are shown in Figure \ref{fig:otu_succession}.} 
\label{fig:ch2_network_kmeans} 
\end{figure}

\begin{figure}[htb!] 
\centering 
\includegraphics[width=0.9\textwidth]{Chapter_3_SWI/Figures/Figure_5_NMDS} 
\caption[Non-metric multidimensional scaling (NMDS) of Bray-Curtis similarity indices based on OTU relative abundance.]{Non-metric multidimensional scaling of Bray-Curtis similarity indices of free-living bacterial community composition based OTU relative abundance. Number labels indicate when the sample was collected in weeks since the initial sampling date in July. Identical number labels indicate three replicate samples collected at the same time, except in week 27 (early January) when replicate samples were collected on two separate days. Not every week was sampled during the study period, as indicated by number labels. Color indicates different groups according to hierarchical clustering with a Bray-Curtis similarity cut-off of 0.5. Ellipses represent the 0.95 confidence interval for each Bray-Curtis similarity group. The time period and range of bacterial production (BP) corresponding to each Bray-Curtis similarity group are shown. } 
\label{fig:ch2_nmds} 
\end{figure}

\section{Discussion}\label{discussion}

We observed a significant decline in free-living bacterial OTU richness as bacterial production increased during the midsummer phytoplankton bloom. Previous studies of temperate systems have also reported diversity maxima in the winter and minima in the summer \citep{Gilbert2012-ta, Ladau2013-ro}. Similar seasonal trends in bacterial OTU richness have been reported at a much lower resolution in Antarctic waters \citep{Ghiglione2012-qm,Georges2014-ng,grwddecm12,Luria2014-dj}. \citet{Gilbert2012-ta} observed relatively gradual changes, reporting that day length and serial day alone explained over 66\% of the observed variance in richness during a 6-year time series study of the English Channel. Likewise, during a six-year study in the Northwest Atlantic Ocean, \citet{El-Swais2015-yx} observed a much longer period of decreased richness during the summer and spring that accompanied an extended period of increased primary production. In contrast, we observed only a brief period (\textasciitilde{}4 weeks) of decreased richness that coincided with increased bacterial production rates. The abbreviated nature of phytoplankton blooms in the WAP along with high-resolution temporal sampling, enabled us to observe this pattern, which contrasts with the multiple annual blooms and/or long periods of sustained growth seen in temperate systems. The results of our study demonstrate close temporal linkages between episodic but biologically significant events like phytoplankton blooms and bacterial production and richness. 

In addition to changes in alpha diversity, we measured temporal variation in the taxonomic composition of free-living bacterial communities. The most abundant winter bacterial taxonomic group in our amplicon sequence data was Pelagibacteraceae, a globally abundant clade that correlates negatively with primary production \citep{Morris2002-xb,Sowell2009-oc}. Several bacterial taxa involved in alternative energetic pathways, e.g. SAR406 (sulfur oxidation; \citealt{wright2014genomic}), \textit{Nitrospina} (nitrite oxidation; \citealt{Spieck2014-sp}), and SAR324 (sulfur oxidation and carbon fixation; \citealt{Sheik2014-yg}) dominated early in the winter, in accordance with previous suggestions that chemolithoautotrophy is an important metabolism contributing to cellular production during the winter \citep{grwddecm12,williams2012metaproteomic,manganelli2009major}. Although we did not sequence archaea in this study, ammonia-oxidizing Thaumarchaeota contribute significantly to chemolithoautotrophy during winter, and overall ammonia oxidation rates are higher in winter than summer \citep{tolar2016contribution}. The seasonality in \textit{Nitrospina} may be related to changes in ammonia oxidation rates. 

While species richness did not change significantly until the phytoplankton bloom in summer, free-living bacterial community composition began to change as early as October, with declining relative abundances of Pelagibacteraceae, SAR406, SAR324, and \textit{Nitrospina}. As the season progressed, increasing competition or top-down control may have caused declines in the relative abundances of these taxa, while increasing water temperatures and therefore increased water column stratification would have prevented replenishment of taxa from deeper waters. The increased relative abundances of taxa like Flavobacteriaceae, \textit{Polaribacter}, Flammeovirgaceae, and Rhodobacteraceae before the summer phytoplankton bloom is somewhat surprising given the low levels of chl \textit{a} and bacterial production during this time period. A brief increase in chl \textit{a} in late October coincided with a decline in richness and an increase in bacterial production. While richness increased when chl \textit{a} subsided, bacterial production remained slightly elevated over the next two months. These pre-summer changes in community composition could reflect a direct response by certain bacterial taxa to increasing sunlight, in addition to low-level and ephemeral increases in phytoplankton. For example, some marine Flavobacteria, including \textit{Polaribacter}, contain proteorhodopsin, a light-driven proton pump that can enhance growth in low-nutrient conditions, while Rhodobactaceae have been shown to carry out aerobic anoxygenic photosynthesis \citep{gonzalez2008genome,Kimura2011-eh,Xing2015-kz,Voget2015-ch}. Whether or not pre-summer changes in bacterial community composition are associated with bacterial feedbacks on phytoplankton growth is not known \citep{Amin2012-tv,Amin2015-pp,Prieto2015-oi,Wang2016-lt}. 

As the summer phytoplankton bloom developed, we observed changes in community composition that corresponded to changes in bacterial production. The relative abundance of free-living \textit{Polaribacter} doubled while that of Rhodobacteraceae tripled during the phytoplankton bloom. We noted that different OTUs within Flavobacteriaceae, including \textit{Polaribacter}, and Rhodobacteraceae peaked at different points during the phytoplankton bloom. This suggests niche specialization by different taxa within these groups, perhaps corresponding to different stages in the successive degradation of phytoplankton-derived organic compounds \citep{Teeling2012-jz,Klindworth2014-ba}. \textit{Polaribacter} occurred in a cluster with only one other OTU (Figure \ref{fig:otu_succession} and \ref{fig:ch2_network_kmeans}), suggesting a strong competitive advantage for \textit{Polaribacter} early in the bloom, in support of the hypothesized role of this taxon \citep{williams2013role}. Rhodobacteraceae OTUs that peaked later in during the bloom may utilize secondary products of decomposing organic matter, such as low molecular weight compounds \citep{Voget2015-ch}. We observed a mid-bloom increase in the relative abundance of Gammaproteobacteria, including \textit{Shewanella} and Colwelliaceae, taxa previously associated with substrate rich environments \citep{Baelum2012-re,Delmont2014-ng}. Early February peaks in the abundance of OTUs classified as Rhodobacteraceae and Colwelliaceae corresponded to almost complete depletion of nitrate and nitrite, an unusual event in the WAP that might have caused phytoplankton to release carbohydrates \citep{hansell2014biogeochemistry}. It is important to note that our focus on the <\SI{3.0}{\micro\meter} fraction of the bacterial community excluded particle-associated bacteria, that may have played an increasingly important role as the phytoplankton bloom developed \citep{riemann2000dynamics}.

We used distributed lag models to statistically test the dynamic relationship between phytoplankton abundance based on chl \textit{a} and bacterial production and abundance during this period. The inclusion of 10- and 20-day lags in our model allowed us to capture a greater portion of the variability in these relationships. In January, peak bacterial production occurred about 20 days after the peak in chl \textit{a}, in agreement with previous studies of Antarctic phytoplankton bloom dynamics \citep{billen1991phytoplankton,ducklow2001seasonal}. A secondary chl \textit{a} peak in mid-January would have presumably also contributed to the peak in bacterial production that occurred about 10 days later.${~}$Bacterial reliance on high molecular weight phytoplankton-derived organic matter that must undergo extracellular hydrolysis prior to bacterial uptake is thought to drive such temporal lags \citep{billen1991phytoplankton,ducklow2001seasonal,lancelot1991modelling,kirchman2001glucose}. Furthermore, some DOM fractions may become available through intermediate trophic processes (e.g. zooplankton sloppy feeding or excretion; \citealt{dsvse12}). The significance of both 10- and 20-day lags may reflect the complex nature of WAP DOM degradation, in which individual bacterial taxa respond within different time frames and specialize in different fractions of the DOM pool \citep{nikrad2014uptake,bowman2015microbial,kim2016decedal}.

We hypothesize that resource supply from phytoplankton-derived carbon was the primary factor driving the summertime shift in free-living bacterial community composition, although biotic interactions and top-down control by grazing and viral lysis likely modulate the patterns that we observed \citep{bird1999uncoupling,Brum2016-ig}. Our measurements did not directly address the underlying mechanisms controlling temporal patterns in bacterial community composition. In particular, we did not directly measure the composition or turnover of organic compounds that we hypothesize contributed to changing bacterial communities. Our study and other similar studies also lack a multi-year dimension to be definitive. Nevertheless, the general pattern that we observed is consistent with the view that increased resource supply of a diverse pool of organic compounds that results from phytoplankton blooms plays a strong role in structuring bacterial communities \citep{Teeling2012-jz,Buchan2014-yh}. 

Taken together, our bacterial production, richness, and community composition data indicate strong trophic coupling between bacteria and phytoplankton. Our results agree with previous reports that WAP bacterial density and production are positively correlated with primary production \citep{Church2003-oj,Ortega-Retuerta2008-pk}, but contrast with earlier reports that the microbial loop is uncoupled from primary producers during spring phytoplankton blooms \citep{Bird1991-sx,bird1999uncoupling,Karl1991-cs,duarte2005experimental}. Although community shifts like those we observed suggest strong potential for degradation of labile phytoplankton-derived organic matter \citep{Landa2014-yg}, WAP bacterial standing stocks and productivity are low relative to primary production (\textasciitilde{}5\% versus 10-20\% global average; \citealt{dsvse12,kim2016decedal}). Despite the longstanding debate about the role of temperature in limiting bacterial production in cold water, several analyses found that temperature is not the principal factor regulating bacterial growth in the Antarctic \citep{dsvse12,thingstad1991bacteria,Kirchman2009-sg}. However, there is growing evidence, including results from this study, that labile organic matter availability is a primary factor controlling bacterial growth and community composition in Antarctica \citep{Kirchman2009-sg,dsvse12,kim2016decedal}. 


