\chapter{Introduction}\label{ch:intro}

\section{An Ocean of Microbes}
 
Ocean covers 70\% of the Earth's surface and each milliliter of seawater contains, on average, $10^{5}$ prokaryotic (Bacterial and Archaeal) cells \citep{Whitman1998-sj,Porter1980-er,Hobbie1977-fh}. With a predicted $3.6 \times 10^{29}$ microbial cells with a total cellular carbon content of \textasciitilde{} $3 \times 10^{17}$ grams, microbes account for most oceanic biomass and play a central role in the global carbon cycle. Marine phytoplankton produce an estimated 45 Tg of organic carbon every year through photosynthesis \citep{Falkowski1998-pw}. Bacterial production and efficiency is largely responsible for determining the fate of this carbon, whether it supports higher trophic levels, is returned directly to the atmosphere as \ce{CO_2}, or is exported to the deep sea via the biological carbon pump \citep{Jiao2010-vq}. Bacteria also support continued primary production via the microbial loop wherein nutrients are regenerated through the remineralization of phytoplankton-derived organic matter \citep{Azam1983-wo}. Marine microbial activities influence \ce{CO_2} levels in the atmosphere, drive biogeochemical cycles in the oceans, and form the foundation of the oceanic foodweb.
 
Describing highly complex marine microbial assemblages is a significant challenge. While microbiologists have not reached a consensus on the definition of species for microbes, microbial communities in the ocean are generally comprised of hundreds to thousands of operational taxonomic units \citep{smhwhnah06,Sunagawa1261359}. Typically, communities consist of a few abundant, often ubiquitous, species and a very large number of less abundant taxa, known as the ``rare biosphere'', which may represent a reservoir of genetic/functional diversity that allows communities to respond rapidly to perturbation \citep{Gibbons2013-fy,smhwhnah06,pedrosalio2006}. Each assemblage may encompass massive metabolic diversity with organisms that fit classical definitions of autotrophs and heterotrophs and a variety of mixotrophs that challenge our traditional understanding of microbial foodwebs \citep{Moran2015-dd}.

Despite widespread implications for ecosystem function, through much of the 20th century microbial oceanographers largely relied on culture-dependent methods to describe the phylogenetic and metabolic diversity of environmental microbes. These methods miss the vast majority of marine microbial diversity, dramatically biasing our early understanding of microbes in the ocean \citep{Staley1985-hf}. In parallel, ``black box'' approaches examined bulk processes, such as primary and bacterial production \citep{karl2007history}. However, increasing capabilities and declining costs for high-throughput DNA sequencing and computation have allowed for the widespread use of molecular approaches \citep{schloss2004;Sunagawa1261359}. In the 1990s, the first bacterial genomes were assembled; as of 2016, more than 30,000 bacterial genomes have been sequenced, providing extensive functional and taxonomic information \citep{Land2015-yf}. Microbial diversity and phylogeny are now frequently inferred from rRNA gene sequences \citep{Woese1987-ps,Pace2009-ck}, while metagenomics provides a snapshot of genomes from communities of microbes, providing additional insights into the functional capabilities of microbes in the environment {venter2004}. Using a combination of traditional and molecular techniques, the ``black box'' of microbial diversity is rapidly being opened, enabling greater understanding of the phylogeny, metabolic potential, and activities of complex microbial assemblages. Despite this progress, the ocean, with its vast scale and remoteness, remains an undersampled and poorly understood environment.

\section{Community Assembly Rules}
 
Perhaps due to limited studies on microbial diversity in the past, most ecological theory was developed based on the abundance and distribution of macrobes, i.e. plants and animals. This trend is shifting as many more studies on microbial diversity in the environment have become available \citep{Barberan2014-pw}. A fundamental aspect of the prediction and understanding of community structure is defining the rules that dictate how species are assembled into communities. \citet{Diamond1975-gh} first used the term ``assembly rules'' to describe how different communities emerge from a common species pool. Niche-based models of community assembly tend to focus on deterministic factors, such as competition and niche differentiation. For a given set of environmental conditions, community development is thought to be convergent, with similar conditions producing similar final community structures. Critics of purely niche-based models argue that many biogeographic patterns can be explained through random distribution alone \citep{Connor1979-go}. The neutral model of community assembly explains community composition as a stochastic balance of birth, death, and migration, assuming functional equivalence for all trophically similar species in the community \citep{hubbell2001unified}. Although this base assumption is troubling to many ecologists, neutral models have successfully reproduced observed species abundance distributions \citep{Enquist2002-fq,Hubbell2006-kd}.
 
The application of either type of model to microbial assemblages is controversial. Microbial communities may defy prevailing ecological theory, due to large population sizes, high dispersal rates, and functional redundancy and previous studies provide conflicting evidence \citep{Fenchel2004-ek}. In the ocean, significant co-occurrence patterns, as well as multi-year time series suggest non-random community assembly, in favor of a niche-based model \citep{fuhrman2006annually,Gilbert2012-ta,El-Swais2015-yx,Eiler2011-jl}. Convergent community assembly due to environmental forcing, i.e. niche processes, has also been observed between replicates in a number of experiments \citep{Kurtz1998-kl,Ayarza2010-gc,Lazzaro2011-ze}. However, other studies have shown divergent assembly, with inoculation of identical replicate environments with the same initial microbial community resulting in different final community structures, suggesting the influence of stochastic forces \citep{pagaling2014community,langenheder2006structure,Roeselers2006-tp}. Given the dominance of bacteria in the environment, understanding the laws governing the abundance and distribution of microbial taxa is essential to understanding ecosystem function.
 
\section{Polar Ecosystems: A Case Study of the Western Antarctic Peninsula (WAP)}

With intra-annual, interannual, and climate change driven decadal variability in ecosystem processes, the WAP system provides a dynamic environment in which to study marine microbial diversity and community assembly. The Palmer Long-Term Ecological Research (LTER) site was established in 1990 with a focus on the pelagic WAP ecosystem, particularly sea ice and upper water column dynamics and related ecological processes, resulting in rich datasets and a relatively well-characterized environment \citep{smithLTER1995}. Every spring, the WAP ecosystem undergoes a rapid transition from less than 9 to over 20 hours of sunlight per day, which drives a cascade of environmental changes, including sea ice retreat and water column stratification. Sea ice melt releases carbon, nutrients, and microbes, possibly influencing microbial community composition through ``seeding''.  Shoaling of the upper mixed layer allows phytoplankton to overcome light limitation, allowing for intense phytoplankton blooms that support a highly productive food web \citep{Venables2013-me,Smetacek2005-tz}. The WAP is also subject to strong interannual variability in physical and biological properties that has been linked to Southern Annular Mode and El Nino-Southern Oscillation cycles \citep{saba2014winter}.
 
On a decadal scale, the WAP is one of the most rapidly warming regions on the planet, with dramatic climate change-driven shifts over the last several decades including a 6\textdegree C increase in midwinter temperatures and a 40\% decline in sea ice extent \citep{dcddghmmmms12,Schofield2010-jj,Stammerjohn2008-nj,Steig2009-nb}. Because numerous WAP ecosystem processes, as well as many species, are profoundly influenced by the annual retreat and advance of sea ice, minor shifts in temperature, which cause declines in sea ice duration and extent, can dramatically impact ecosystem function \citep{Moline2008-pc}. Indeed, concurrent declines of Ad\'{e}lie penguin populations \citep{Ducklow2007-ns,Fraser1986-ps} and diatom and krill stocks in northern regions \citep{aspr04,mddfmss09}, as well as shifts in the dominant phytoplankton species from diatoms to cryptophytes \citep{Moline2004-ma} have been reported. However, little is known about WAP bacterial and archaeal community compositional and functional responses to natural variation and warming-induced changes, including variability in phytoplankton dynamics.
 
This thesis focuses on the stochastic and deterministic factors that structure microbial communities, especially heterotrophic bacteria, against a backdrop of seasonal transitions, interannual variability, and long-term climate changes in the coastal waters of the WAP. The Palmer LTER project provided rich contextual data and an intellectual framework in which to embed my dissertation research. My work was primarily carried out at Palmer Station, with a relatively large team of assistants, often in collaboration with the LTER, while data generation and DNA sequencing was carried out at the Marine Biological Laboratory (MBL) in Woods Hole, MA.
 
I begin, in Chapter \ref{ch:mirada}, with a characterization of eukaryotic, archaeal, and bacterial diversity at different sites (northern, southern, inshore, and offshore) along the peninsula, using 16S rRNA gene V6 hypervariable region amplicon sequence data collected through the Microbial Inventory Research Across Diverse Aquatic (MIRADA) LTERs project. This work was initiated by Linda Amaral-Zettler, in collaboration with Hugh Ducklow, prior to my arrival at Brown University and MBL. A climatic gradient running parallel to the WAP allows for a comparison between warmer, maritime conditions in the north, and colder, drier conditions in the south. While most samples were collected during the austral summer, a limited number of winter samples indicated intriguing temporal differences in microbial community composition \citep{Luria2014-dj}.
 
For Chapter \ref{ch:swi}, I focused on these seasonal differences, examining changes in bacterial community composition and richness over a nine-month period that spanned winter through the end of summer. A number of changes in bacterial diversity and community composition corresponded with an intense summer phytoplankton bloom. I hypothesized that these changes were driven by the release of phytoplankton-derived dissolved organic matter (DOM).
 
I tested the hypothesis that phytoplankton-derived DOM is an important driver of bacterial community structure in Chapter \ref{ch:dom}, through a series of DOM-addition mesocosm experiments. I found that DOM addition affected the bacterial community in different ways at different points during the season. Some of the resulting changes did indeed reflect those that occur during a phytoplankton bloom, while others differed substantially from trends seen in the environment.
 
In Chapter \ref{ch:lter}, I expanded my time series analysis with bacterial community composition data that span four Palmer LTER seasons, with a focus on repeatability in bacterial seasonal succession. While some aspects of succession were consistent from year to year, interannual variation was substantial.
 
Finally, in Chapter \ref{ch:seaice}, I conducted sea ice mesocosm experiments, in which I tested the effects of both filtered and unfiltered sea ice meltwater to examine the relative influences of both DOM and nutrient injections and of microbial seeding.
 
This body of work significantly increases our knowledge of microbial community structure in the WAP. It includes the first comparative study of microbial ecology along the climatic gradient running parallel to the Antarctic Peninsula, a fine-scale analysis of bacterial succession across the austral winter-to-summer seasonal transition, and a broader analysis of bacterial succession across multiple austral summers. I have explored the factors that drive bacterial community assembly, particularly the availability of phytoplankton-derived DOM, advancing our understanding of the limits on microbial heterotrophic activity and carbon recycling in the WAP. Throughout, I have considered the application of ecological theory to marine microbes.
